app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt

```diff
@@ -55,11 +56,20 @@ class SyncIndicatorViewController @Inject constructor(
     val showSyncIndicatorState = Observables
         .combineLatest(screenCreated, syncProgress, configProvider)
         { _, stateStream, config ->
-          val indicatorState = when (stateStream.lastSyncProgress!!) {
+          when (stateStream.lastSyncProgress!!) {
```

Comment:
Are we sure that `lastSyncProgress` will not be null here? Maybe we could add a null branch to this when block and set some default UI state.

Reply:
The `null` check already happens in the `syncProgress` stream.
Check this:

`    val syncProgress = lastSyncedStateStream.filter { it.lastSyncProgress != null }
`